app/crud/sync_db/tile_cache_assets.py

```diff
@@ -1,112 +1,154 @@
-from typing import Any, Dict, List, Optional, Tuple
+from typing import Any, Dict, List, Optional
 
 from cachetools import TTLCache, cached
 from fastapi.logger import logger
 
 from ...application import get_synchronous_db
+from ...models.enumerators.tile_caches import TileCacheType
 
 
-# @cached(cache=TTLCache(maxsize=15, ttl=900))
-def get_datasets(asset_type: str) -> List[str]:
+@cached(cache=TTLCache(maxsize=1, ttl=900))
+def get_all_tile_caches():
+
+    tile_caches: Dict[str, List] = {e.value: list() for e in list(TileCacheType)}
+
     with get_synchronous_db() as db:
         rows = db.execute(
-            """SELECT DISTINCT
-                    dataset
-                   FROM assets
-                   WHERE asset_type = :asset_type
-                    AND status = 'saved'""",
-            {"asset_type": asset_type},
+            f"""SELECT DISTINCT
+                    assets.asset_type,
+                    versions.dataset,
+                    versions.version,
+                    assets.creation_options->'implementation' as implementation,
+                    versions.is_latest,
+                    assets.fields,
+                    assets.creation_options->'min_zoom' as min_zoom,
+                    assets.creation_options->'max_zoom' as max_zoom
+                   FROM versions
+                    JOIN assets
+                    ON versions.dataset = assets.dataset
+                        AND versions.version = assets.version
+                   WHERE assets.asset_type IN {str(tuple([e.value for e in list(TileCacheType)])).replace('"', "'")}
+                    AND assets.status = 'saved'"""
         ).fetchall()
+    if not rows:
+        # tile_caches = []
+        logger.warning("There are no tile caches registered with the API.")
+    # else:
+    # tile_caches = rows
+
+    for row in rows:
+        tile_caches[row[0]].append(
+            {
+                "dataset": row[1],
+                "version": row[2],
+                "implementation": row[3],
+                "is_latest": row[4],
+                "fields": row[5],
+                "min_zoom": row[6],
+                "max_zoom": row[7],
+            }
+        )
```

Comment:
I think you can access these as row["dataset"] and even row.dataset, which might be better in case someone changes the schema or query. Just a suggestion, certainly not a blocker.